graph/ForceDirectedGraph/ForceDirectedGraph.tsx

```diff
@@ -0,0 +1,264 @@
+    }
+  };
+
+  // If the graph is not loaded, display a loading message
+  if (!load) {
+    return (
+      <View style={styles.container}>
+        <Text>Loading...</Text>
+      </View>
+    );
+  }
+
+  const coordX = (node: Node): number => {
+    if (dragEnabled || clickedNodeID === node.id) {
+      return node.x + totalOffset.x;
+    }
+    return node.x;
+  };
+
+  const coordY = (node: Node): number => {
+    if (dragEnabled || clickedNodeID === node.id) {
+      return node.y + totalOffset.y;
+    }
+    return node.y;
+  };
+
+  const circleFill = (node: Node): string => {
+    if (node.id === constrainedNodeId) {
+      return CONSTRAINED_NODE_COLOR;
+    }
+    return DEFAULT_NODE_COLOR;
+  };
+  const LINES = links.map((link) => (
+    <Line
+      key={link.source + link.target}
+      x1={coordX(nodes.find((node) => node.id === link.source) ?? {x: screenWidth/2, y: screenHeight/2, id: link.source, dx: 0, dy: 0})}
+      y1={coordY(nodes.find((node) => node.id === link.source) ?? {x: screenWidth/2, y: screenHeight/2, id: link.source, dx: 0, dy: 0})}
+      x2={coordX(nodes.find((node) => node.id === link.target) ?? {x: screenWidth/2, y: screenHeight/2, id: link.target, dx: 0, dy: 0})}
+      y2={coordY(nodes.find((node) => node.id === link.target) ?? {x: screenWidth/2, y: screenHeight/2, id: link.target, dx: 0, dy: 0})}
+      stroke={DEFAULT_LINK_COLOR}
+    />
+  ));
+
+  const CIRCLES = nodes.map((node) => (
+    <Circle
+      key={node.id}
+      cx={coordX(node)}
+      cy={coordY(node)}
+      r={sizes.get(node.id) ?? DEFAULT_NODE_SIZE}
+      fill={circleFill(node)}
+      onPressIn={() =>
+        handlePressIn(() => {
+          pressStartRef.current = Date.now();
+          setClickedNodeID(node.id);
+        })
+      }
+      onPressOut={() =>
+        handlePressOut(() => {
+          setClickedNodeID(DEFAULT_CLICKED_NODE_ID);
+        })
+      }
+    />
+  ));
+
+  const GRAPH_CANVAS = (
+    <View style={styles.container}>
+      <Svg width={screenWidth} height={screenHeight}>
+        <G scale={scale} originX={screenWidth / 2} originY={screenHeight / 2}>
+          {LINES}
+          {CIRCLES}
+        </G>
+      </Svg>
+    </View>
+  );
+
+  const PAN_GESTURE_HANDLER = (
+    <PanGestureHandler
+      ref={panRef}
+      onGestureEvent={handlePanGestureEvent}
+      onHandlerStateChange={handlePanHandlerStateChange}
+      minPointers={1}
+      maxPointers={1}
+      simultaneousHandlers={pinchRef}
+    >
+      {GRAPH_CANVAS}
+    </PanGestureHandler>
+  );
+
+  const PINCH_GESTURE_HANDLER = (
+    <PinchGestureHandler
+      ref={pinchRef}
+      onGestureEvent={handlePinchGestureEvent}
+      onHandlerStateChange={handlePinchHandlerStateChange}
+      simultaneousHandlers={panRef}
+    >
+      {PAN_GESTURE_HANDLER}
+    </PinchGestureHandler>
+  );
+  return (
+    <GestureHandlerRootView style={styles.container}>
+      {PINCH_GESTURE_HANDLER}
+    </GestureHandlerRootView>
+  );
+};
```

Comment:
I like your personal choice of modularizing everything, but if there are many parent components, it becomes a bit hard to visually see the hierarchy of components. I suggest returning plain pinch and gesture handlers together with the graph canvas in the final return statement. Only Circles and Lines should be defined externally. However, that's just my opinion, either way works.